### PR TITLE
Adding roxy property to ignore certain files or folders

### PR DESCRIPTION
**Pull request by miguelrgonzalez**
Specially when working with node and bower there are certain folder we don't want to be deployed into the server.

### CODE REVIEW

**Comment by grtjn** [comment]
Have you considered keeping those outside src/?


Sent from Samsung Mobile

**Comment by miguelrgonzalez** [comment]
Yes, I did.
But I prefer to have source code inside the src folder.
Plus having bower.js, gulp.js there also makes sense.

File: deploy/default.properties [hunk 1]
```diff
@@ -204,3 +204,6 @@ mlcp-vmargs=-Xmx512m
 # Temporary fix to support MarkLogic 8 EA2
 #
 evaler-port=7998
+
+# folde to ignore when deploying into modules database
+ignore-folders=.*/src/(node_modules|.tmp)/.*
```
Comment (paxtonhare) [inline deploy/default.properties]: Move this into build.sample.properties as a comment explaining how to use it. I don't think this should be the default for everyone.

File: deploy/lib/server_config.rb [hunk 2]
```diff
@@ -1130,12 +1130,14 @@ def deploy_src
...
     load_js_as_binary = @properties['ml.load-js-as-binary']
     load_css_as_binary = @properties['ml.load-css-as-binary']
+    folders_to_ignore = @properties['ml.ignore-folders']
 
     modules_databases.each do |dest_db|
       ignore_us = []
       ignore_us << "^#{test_dir}.*$" unless test_dir.blank? || deploy_tests?(dest_db)
       ignore_us << "^#{app_config_file}$"
       ignore_us << "^#{test_config_file}$"
+	  ignore_us << "^#{folders_to_ignore}$"
```
Comment (paxtonhare) [inline deploy/lib/server_config.rb]: This should only append if folders_to_ignore is not blank.

Comment (paxtonhare) [inline deploy/lib/server_config.rb]: Also, check your usage of tabs versus spaces for consistency.

**Comment by paxtonhare** [comment]
@miguelrgonzalez I like this and think it's useful. I added some inline comments that I'd like to see fixed before we merge it.

**Comment by miguelrgonzalez** [comment]
Hi Paxton,
will do the requested changes and fix my commit.

**Comment by paxtonhare** [comment]
Awesome. Thanks for submitting it.

On Mon, Jan 12, 2015 at 10:08 AM, Miguel Rodríguez <notifications@github.com

**Comment by dmcassel** [comment]
+1 on this, once Paxton's comments are addressed

**Comment by grtjn** [comment]
One sec before merging. I think it would be better to push this into the dev branch.

@miguelrgonzalez would it be a lot of trouble to ask you to redo your changes based on the dev branch, and do a PR against that one?

**Comment by dmcassel** [comment]
Good catch, @grtjn

**Comment by miguelrgonzalez** [comment]
No problem, it's just a couple of lines really.

**Comment by grtjn** [comment]
Replaced by PR #337..

**Pull request closed without merging**